File: scraper.js
const fs = require('fs');

class LotoScraper {
    constructor(filename = 'historico_loto.csv') {
        this.filename = filename;
    }

    collectData() {
        console.log("🔎 [Scraper] Verificando fuente de datos y sorteos...");
        const datos = [];
        if (fs.existsSync(this.filename)) {
            const fileContent = fs.readFileSync(this.filename, 'utf-8');
            const lines = fileContent.trim().split('\n');
            const headers = lines[0].split(',');
            
            for (let i = 1; i < lines.length; i++) {
                const values = lines[i].split(',');
                if (values.length === headers.length) {
                    const row = {};
                    for (let j = 0; j < headers.length; j++) {
                        row[headers[j].trim()] = values[j].trim();
                    }
                    datos.push(row);
                }
            }
            console.log(`✅ [Scraper] ${datos.length} sorteos cargados desde el historial local.`);
        } else {
            console.log("⚠️ [Scraper] No se encontró historial. Se recomienda recopilar datos inicialmente.");
        }
        return datos;
    }

    mockFetchNewDraw(lastDrawNum) {
        const numbers = new Set();
        while(numbers.size < 6) {
            numbers.add(Math.floor(Math.random() * 46));
        }
        const sortedNumbers = Array.from(numbers).sort((a,b) => a - b);
        const plus = Math.floor(Math.random() * 10);
        
        const pad = (num) => String(num).padStart(2, '0');
        
        return {
            'Sorteo': String(lastDrawNum + 1),
            'Fecha': new Date(Date.now() + 3*24*60*60*1000).toISOString().split('T')[0],
            'B1': pad(sortedNumbers[0]),
            'B2': pad(sortedNumbers[1]),
            'B3': pad(sortedNumbers[2]),
            'B4': pad(sortedNumbers[3]),
            'B5': pad(sortedNumbers[4]),
            'B6': pad(sortedNumbers[5]),
            'Plus': pad(plus)
        };
    }
}

module.exports = LotoScraper;

File: scraper.py
import csv
import os
import random
from datetime import datetime, timedelta

class LotoScraper:
    def __init__(self, filename="historico_loto.csv"):
        self.filename = filename
        
    def collect_data(self):
        """
        Lee los datos historicos existentes. 
        En un escenario real iteraría librerias web (como BeautifulSoup y requests)
        para extraer sorteos nuevos de sitios como TuJugada.com.ar o la web oficial
        y los guardaría en el CSV antes de retornarlos.
        """
        print("🔎 [Scraper] Verificando fuente de datos y sorteos...")
        
        datos = []
        if os.path.exists(self.filename):
            with open(self.filename, mode='r', encoding='utf-8') as f:
                reader = csv.DictReader(f)
                for row in reader:
                    datos.append(row)
            print(f"✅ [Scraper] {len(datos)} sorteos cargados desde el historial local.")
        else:
            print("⚠️ [Scraper] No se encontró historial. Se recomienda recopilar datos inicialmente.")
        
        return datos

    def mock_fetch_new_draw(self, last_draw_num):
        """
        Función para simular la llegada de un nuevo sorteo.
        """
        numeros = random.sample(range(0, 46), 6)
        numeros.sort()
        plus = random.randint(0, 9)
        nuevo_sorteo = {
            'Sorteo': str(last_draw_num + 1),
            'Fecha': (datetime.now() + timedelta(days=3)).strftime("%Y-%m-%d"),
            'B1': f"{numeros[0]:02d}", 'B2': f"{numeros[1]:02d}", 'B3': f"{numeros[2]:02d}",
            'B4': f"{numeros[3]:02d}", 'B5': f"{numeros[4]:02d}", 'B6': f"{numeros[5]:02d}",
            'Plus': f"{plus:02d}"
        }
        return nuevo_sorteo
